provider/src/main/kotlin/org/gradle/kotlin/dsl/accessors/CodeGenerator.kt

```diff
@@ -22,142 +22,142 @@ import org.jetbrains.kotlin.lexer.KtTokens
 
 internal
 fun ProjectSchema<TypeAccessibility>.forEachAccessor(action: (String) -> Unit) {
-    extensions.forEach { (name, type) ->
-        extensionAccessorFor(name, type)?.let(action)
+    extensions.map(::typedAccessorSpecFor).forEach { spec ->
+        extensionAccessorFor(spec)?.let(action)
     }
-    conventions.forEach { (name, type) ->
-        if (name !in extensions) {
-            conventionAccessorFor(name, type)?.let(action)
+    conventions.map(::typedAccessorSpecFor).forEach { spec ->
+        if (spec.name.original !in extensions) {
+            conventionAccessorFor(spec)?.let(action)
         }
     }
-    configurations.forEach { name ->
-        configurationAccessorFor(name)?.let(action)
+    configurations.map(::accessorNameSpecFor).forEach { spec ->
+        configurationAccessorFor(spec)?.let(action)
     }
 }
 
 private
-fun extensionAccessorFor(name: String, typeAccess: TypeAccessibility): String? =
-    codeForExtension(name) {
-        when (typeAccess) {
-            is TypeAccessibility.Accessible   -> accessibleExtensionAccessorFor(name, typeAccess.type)
-            is TypeAccessibility.Inaccessible -> inaccessibleExtensionAccessorFor(name, typeAccess)
+fun extensionAccessorFor(spec: TypedAccessorSpec): String? =
+    codeForAccessor(spec.name) {
+        when (spec.typeAccess) {
+            is TypeAccessibility.Accessible -> accessibleExtensionAccessorFor(spec.name, spec.typeAccess.type)
+            is TypeAccessibility.Inaccessible -> inaccessibleExtensionAccessorFor(spec.name, spec.typeAccess)
         }
     }
 
 
 private
-fun accessibleExtensionAccessorFor(name: String, type: String): String =
+fun accessibleExtensionAccessorFor(name: AccessorNameSpec, type: String): String =
```

Comment:
:x: Same as above. Minimize the number of references to name using `...: String = name.run { ...`.